solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -82,90 +84,76 @@ public DocTransformer create(String name, SolrParams params,
 
     /**
      * @param name
-     *          Name of the field to be added in a document representing the
-     *          feature vectors
+     *          Name of the field to be added in a document representing the feature vectors
      */
     public FeatureTransformer(String name, SolrParams params,
         SolrQueryRequest req) {
       this.name = name;
       this.params = params;
       this.req = req;
+      this.ltrHelper = new LTRRerankHelper(req, params);
     }
 
     @Override
     public String getName() {
       return name;
     }
 
-    @Override
-    public void setContext(ResultContext context) {
-      super.setContext(context);
+    private boolean setSearcher(ResultContext context) {
       if (context == null) {
-        return;
+        return false;
       }
-      if (context.getRequest() == null) {
-        return;
+      if (context.getSearcher() == null) {
+        return false;
       }
-
       searcher = context.getSearcher();
-      if (searcher == null) {
+      return true;
+    }
+
+    private void setupRerankModel(String featureStoreName) {
+      if (featureStoreName == null) {
+        featureStoreName = ltrHelper.getDefaultFeatureStoreName();
+      }
+
+      final ManagedFeatureStore fr = ltrHelper.getFeatureStore();
+      FeatureStore store = null;
+      try{
+        store = fr.getFeatureStore(featureStoreName);
+      }catch (final Exception e) {
+        throw new SolrException(ErrorCode.BAD_REQUEST,
+            "retrieving the feature store "+featureStoreName, e);
+      }
+      final LoggingModel loggingModel = new LoggingModel(featureStoreName, store.getFeatures());
+      reRankModel = ltrHelper.getRerankModel(loggingModel);
+      reRankModel.setOriginalQuery(context.getQuery());
+  }
+
+    @Override
+    public void setContext(ResultContext context) {
+      super.setContext(context);
+      if (! setSearcher(context)) {
         throw new SolrException(
             org.apache.solr.common.SolrException.ErrorCode.BAD_REQUEST,
             "searcher is null");
       }
       leafContexts = searcher.getTopReaderContext().leaves();
 
       // Setup ModelQuery
-      reRankModel = (ModelQuery) req.getContext().get(CommonLTRParams.MODEL);
-      resultsReranked = (reRankModel != null);
-      String featureStoreName = params.get(CommonLTRParams.STORE);
+      
+      reRankModel = ltrHelper.getModelFromContext();
+      resultsReranked = ltrHelper.hasModelInContext();
```

Comment:
I'd like to minimize the size of the ltrHelper class.  Right now it seems to have lots of things in it, making it hard for me to really know how/when to use it vs checking params.get() directly.
Let;s remove hasModelInContext() and just check the reRankModel != null like before